core/src/templates/components/card/card.twig

```diff
@@ -64,28 +155,30 @@
     {%- block block_card_headline  %}
       {%- if card_headline is not empty %}
         <h2>
-          {%- if card_headline_link and allow_links -%}
```

Comment:
If you want to merge this PR into master you will have to maintain this variable. We can change it in V6 and mark it deprecated here in master but we are not allowed to change it before the next major version.